src/lib/yacc/parser.rs

```diff
@@ -236,6 +236,9 @@ impl YaccParser {
 
     fn parse_rule(&mut self, mut i: usize) -> YaccResult<usize> {
         let (j, rn) = try!(self.parse_name(i));
+        if self.ast.start.is_none() {
+            self.ast.start = Some(rn.clone());
```

Comment:
Just to double check: I assume `rn` is a list, so the first element really is the first rule in the grammar.

Reply:
`rn` is a String (the rule name).

Reply:
Ah, I see now. The first rule that gets parsed becomes the start rule if start-rule is not already set. Guess I didn't read the code carefully enough. :)

Reply:
:)